File: es2/es2b.py
#!/usr/bin/env python3

with open("input.txt", "r") as f:
    rows = f.read().split("\n")

def distance(a, b):
    if len(a) != len(b):
        return 26

    d = 0
    for i in range(len(a)):
        if a[i] != b[i]:
            d += 1
    return d

w1, w2 = None, None
mind = 26
rowcount = len(rows)-1
for i in range(rowcount-1):
    for j in range(i+1, rowcount):
        d = distance(rows[i], rows[j])
        if d < mind:
            mind = d
            w1 = i
            w2 = j

l = len(rows[i])
common = []
for i in range(l):
    if rows[w1][i] == rows[w2][i]:
        common.append(rows[w1][i])

print("".join(common))

File: es2/es2b.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func distance(w1, w2 string) int {
	d := 0
	for i := 0; i < len(w1); i++ {
		if w1[i] != w2[i] {
			d++
		}
	}
	return d
}

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer input.Close()

	var lines []string
	scanner := bufio.NewScanner(input)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	mind := 26
	var w1 string
	var w2 string
	for i := 0; i < len(lines)-1; i++ {
		for j := i + 1; j < len(lines); j++ {
			d := distance(lines[i], lines[j])
			if mind > d {
				mind = d
				w1 = lines[i]
				w2 = lines[j]
			}
		}
	}

	for i := 0; i < len(w1); i++ {
		if w1[i] == w2[i] {
			fmt.Printf("%c", w1[i])
		}
	}
	fmt.Println()
}
